rust/protocol/src/sealed_sender.rs

```diff
@@ -879,8 +1019,214 @@ mod sealed_sender_v2 {
 
         HKDF::new(3)?.derive_secrets(&agreement_key_input, LABEL_DH_S, AUTH_TAG_LEN)
     }
+
+    #[test]
+    fn test_agreement_and_authentication() -> Result<()> {
+        // The sender and recipient each have a long-term identity key pair.
+        let sender_identity = IdentityKeyPair::generate(&mut rand::thread_rng());
+        let recipient_identity = IdentityKeyPair::generate(&mut rand::thread_rng());
+
+        // Generate random bytes used for our multi-recipient encoding scheme.
+        let m: [u8; MESSAGE_KEY_LEN] = (&mut rand::thread_rng()).gen();
+        // Derive an ephemeral key pair from those random bytes.
+        let ephemeral_keys = DerivedKeys::calculate(&m);
+        let ephemeral_public_key = ephemeral_keys.e.public_key;
+
+        // Encrypt the ephemeral key pair.
+        let sender_c_0: [u8; MESSAGE_KEY_LEN] = apply_agreement_xor(
+            &ephemeral_keys.e,
+            recipient_identity.public_key(),
+            Direction::Sending,
+            &m,
+        )?
+        .as_ref()
```

Comment:
I don't think you need this `as_ref()` at all; [Deref](https://doc.rust-lang.org/std/ops/trait.Deref.html#more-on-deref-coercion) should take care of it.

Reply:
Looks to be so!